official/vision/beta/projects/yolo/losses/yolo_loss.py

```diff
@@ -0,0 +1,654 @@
+
+    # Search all predictions against ground truths to find mathcing boxes for
+    # each pixel.
+    _, _, iou_max, _ = self._search_pairs(
+        pred_boxes, pred_classes, boxes, classes, scale=scale, yxyx=True)
+
+    # Find the exact indexes to ignore and keep.
+    ignore_mask = tf.cast(iou_max < self._ignore_thresh, pred_boxes.dtype)
+    iou_mask = iou_max > self._ignore_thresh
+
+    if not smoothed:
+      # Ignore all pixels where a box was not supposed to be predicted but a
+      # high confidence box was predicted.
+      obj_mask = true_conf + (1 - true_conf) * ignore_mask
+    else:
+      # Replace pixels in the tre confidence map with the max iou predicted
+      # with in that cell.
+      obj_mask = tf.ones_like(true_conf)
+      iou_ = (1 - self._objectness_smooth) + self._objectness_smooth * iou_max
+      iou_ = tf.where(iou_max > 0, iou_, tf.zeros_like(iou_))
+      true_conf = tf.where(iou_mask, iou_, true_conf)
+
+    # Stop gradient so while loop is not tracked.
+    obj_mask = tf.stop_gradient(obj_mask)
+    true_conf = tf.stop_gradient(true_conf)
+    return true_conf, obj_mask
+
+  def __call__(self, true_counts, inds, y_true, boxes, classes, y_pred):
+    """Call function to compute the loss and return the total loss as 
+    well as the loss for each detection mask on a given FPN level. 
+    
+    Args: 
+      true_counts: `Tensor` of shape [batchsize, height, width, num_anchors] 
+        represeneting how many boxes are in a given pixel [j, i] in the 
+        output map.
+      inds: `Tensor` of shape [batchsize, None, 3] indicating the location 
+        [j, i] that a given box is associatied with in the FPN prediction 
+        map. 
+      y_true: `Tensor` of shape [batchsize, None, 8] indicating the actual box 
+        associated with each index in the inds tensor list.
+      boxes: `Tensor` of shape [batchsize, None, 4] indicating the original 
+        ground truth boxes for each image as they came from the decoder used
+        for bounding box search. 
+      classes: `Tensor` of shape [batchsize, None, 1] indicating the original 
+        ground truth classes for each image as they came from the decoder used
+        for bounding box search.
+      y_pred: `Tensor` of shape [batchsize, height, width, output_depth] 
+        holding the models output at a specific FPN level.
+
+    Return:
+      loss: `float` for the actual loss.
+      box_loss: `float` loss on the boxes used for metrics.
+      conf_loss: `float` loss on the confidence used for metrics.
+      class_loss: `float` loss on the classes used for metrics.
+      avg_iou: `float` metric for the average iou between predictions 
+        and ground truth.
+      avg_obj: `float` metric for the average confidence of the model 
+        for predictions.
+    """
+    (loss, box_loss, conf_loss, class_loss, mean_loss, iou, pred_conf, ind_mask,
+     grid_mask) = self.call(true_counts, inds, y_true, boxes, classes, y_pred)
```

Comment:
Why there is __call__ and call? you can make the abstrat function a private function with an initial '_' and call it _compute_loss or something.